tests/test_flow/test_persistence_random.py

```diff
@@ -343,11 +382,26 @@ def new():
             self._child_counts_by_entity_name[entity_name] = 0
             return entity_name
 
-        if self._random_bool(p_true=1 - self._config.p_multi_out):
+        include_multiple_entities = self._random_bool(p_true=self._config.p_multi_out)
+        if not include_multiple_entities:
             return new()
-        else:
+
+        include_three_entities = self._random_bool(
+            p_true=self._config.p_three_out_given_multi
+        )
+        if not include_three_entities:
             return f"{new()}, {new()}"
 
+        return self._rng.choice(
+            [
+                lambda: f"{new()}, {new()}, {new()}",
```

Comment:
nit: so many brackets 😵. Can you add a small example as a comment after each choice? I mean something like `a, b, c`, `(a, b), c`, etc.